platform/admin/admin-configuration-configupdater/src/main/java/org/codice/ddf/admin/configuration/ConfigurationUpdater.java

```diff
@@ -0,0 +1,322 @@
+        return;
+      }
+    }
+
+    // Routine property updates for tracked files - write to disk if necessary (4)
+    writeIfNecessary(
+        appropriatePid, fileFromConfigAdmin, context.getSanitizedProperties(), cachedConfigData);
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * <p>Remove config files from etc when a delete occurs but only if felix is tracking the file.
+   *
+   * @param pid the pid pointing to the configuration to delete.
+   */
+  @Override
+  public void handleDelete(String pid) throws IOException {
+    try {
+      AccessController.doPrivileged(
+          (PrivilegedExceptionAction<Object>)
+              () -> {
+                doHandleDelete(pid);
+                return null;
+              });
+    } catch (PrivilegedActionException e) {
+      // Only caught if the action threw a CHECKED exception, thus it MUST be an IOException.
+      throw (IOException) e.getCause();
+    }
+  }
+
+  private void doHandleDelete(String pid) throws IOException {
+    CachedConfigData cachedConfigData = pidDataMap.remove(pid);
+    File fileFromCache = (cachedConfigData != null) ? cachedConfigData.getFelixFile() : null;
+    if (fileFromCache != null && fileFromCache.exists()) {
+      LOGGER.debug("Deleting file because config was deleted for pid [{}]", pid);
+      try {
+        Files.delete(fileFromCache.toPath());
+        SecurityLogger.audit("Removing a deleted config [{}]", fileFromCache.toPath());
+      } catch (IOException e) {
+        LOGGER.debug(
+            format("Problem deleting config file [%s]: ", fileFromCache.getAbsolutePath()), e);
+        // Synchronous with config admin, so we can report the failure to the UI this way
+        throw e;
+      }
+    }
+  }
+
+  private void writeIfNecessary(
+      String appropriatePid,
+      File dest,
+      Dictionary<String, Object> configState,
+      CachedConfigData cachedData)
+      throws IOException {
+    if (dest != null && dest.exists()) {
+      processPasswords(appropriatePid, configState, this::encryptValue);
+      if (!cachedData.equalProps(configState)) {
+        try (OutputStream outputStream = new BufferedOutputStream(new FileOutputStream(dest))) {
+          getAppropriateStrategy(dest.toPath()).write(outputStream, configState);
+        } catch (IOException e) {
+          LOGGER.debug("Writing to config file failed: ", e);
+          SecurityLogger.audit("Failure to update config file [{}]", dest.getAbsolutePath());
+          throw e;
+        }
+        SecurityLogger.audit("Updating config file [{}]", dest.toPath());
+        cachedData.setProps(configState);
+      }
+    }
+  }
+
+  /**
+   * {@link EncryptionService} is whitelisted, so we cannot add this method onto the interface and
+   * impl until a major release.
+   */
+  private String encryptValue(String plaintextValue) {
+    Matcher m = ENC_PATTERN.matcher(plaintextValue);
+    if (m.find()) {
+      LOGGER.debug("Password already encrypted");
```

Comment:
❓ Are those traces turned on by default? If so then this trace would be noisy.

Reply:
Changed level to `trace` instead of `debug` and no they are not.